unittest/gunit/keyring_vault/vault_parser-t.cc

```diff
@@ -209,6 +210,18 @@ namespace keyring__vault_parser_unittest
     EXPECT_TRUE(vault_parser.parse_key_data(payload, &key));
   }
 
+  TEST_F(Vault_parser_test, ParsePayloadThatsGarbage)
+  {
+    uchar garbage[101000];
+    my_rand_buffer(garbage, sizeof(garbage));
+    Secure_string payload(reinterpret_cast<char*>(garbage));
```

Comment:
There is a non-zero chance that `garbage` won't have any zero bytes in it. So, this is a potential memory overrun.
`Secure_string payload(reinterpret_cast<char*>(garbage), sizeof(garbage));` may be?

Reply:
nice catch